network/WebSocket/WebSocketConnection.cpp

```diff
@@ -515,12 +517,40 @@ namespace awsiotsdk {
                 canonical_query_string.append("&");
                 canonical_query_string.append(X_AMZ_SECURITY_TOKEN);
                 canonical_query_string.append("=");
-                canonical_query_string.append(aws_session_token_);
+                std::string encoded_string = std::string(aws_session_token_);
+                UrlEncode(encoded_string, NOT_ENCODED_CHARS);
+                canonical_query_string.append(encoded_string);
             }
             AWS_LOG_DEBUG(WEBSOCKET_WRAPPER_LOG_TAG, "CompletedCanonicalQuery: %s", canonical_query_string.c_str());
             return ResponseCode::SUCCESS;
         }
 
+        void WebSocketConnection::UrlEncode(std::string &string,
```

Comment:
Same here, util::String instead of std::string and util::Vector instead of std::vector.